### ISSUE TITLE
[BUG] Unable to throw a ValidationApiException exception

### ISSUE THREAD

**Issue by marcominerva** [reactiveui/refit#1376]
I have a Web API that returns a `ProblemDetails` with the `application/problem+json` content type:

![image](https://user-images.githubusercontent.com/3522534/173522324-4fd28ff7-d370-436b-9aaa-8ec741d6ec7b.png)

Then, I have the following Refit interface:

```
public interface IPeopleServiceApi
{
    [Post("/people")]
    Task<ApiResponse<Person>> GetAsync(Person person);
}
```

If I try to call this API from Refit, when there is a validation problem, in the `Error` property of the `ApiResponse` I see a `ValidationApiException` object:

![image](https://user-images.githubusercontent.com/3522534/173523892-2187d9b0-7ea3-4877-b4e0-0bfe402b802d.png)

However, if I call `EnsureSuccessStatusCodeAsync`, the exception that is raised is `ApiException`, not `ValidationApiException`. Moreover, the `HasContent` property of the exception is _false_:

![image](https://user-images.githubusercontent.com/3522534/173525516-8292c69e-67c9-4848-bce0-63ecc1248d74.png)

If, instead, I change the interface to directly return a `Person `instead of an `ApiResponse<Person>`, when I call the API, a `ValidationApiException` is thrown, with the `Content `correctly set.

**Steps To Reproduce**

1. Invoke an API that return a problem details with the `application/problem+json` content type
2. Define a method in the interface for Refit that returns an _ApiReponse<T>_ for this API
3. Invoke the API from the client
4. Call the `ApiResponse.EnsureSuccessStatusCodeAsync` method
5. An `ApiException` is thrown

**Expected behavior**
A `ValidationApiException` should be thrown

**Environment**
- OS:       Windows 11
- Device:   Desktop
- Version:   6.3.2

**Comment by marcominerva** [comment]
After some digging, it seems that the issue depends on this method: https://github.com/reactiveui/refit/blob/246ee8d9989c29092fdaba6821d5b3098a5ccf9e/Refit/ApiResponse.cs#L85

```
public async Task<ApiResponse<T>> EnsureSuccessStatusCodeAsync()
{
    if (!IsSuccessStatusCode)
    {
        var exception = await ApiException.Create(response.RequestMessage!, response.RequestMessage!.Method, response, Settings).ConfigureAwait(false);

        Dispose();

        throw exception;
    }

    return this;
}
```

If there was an error in the response, the `ApiException.Create()` method is called to create the **Exception** object. However, as this point `response.Content` has already been read and disposed, because the `ApiException.Create()` method has already been called when the `ApiResponse<T>` object has been created (to populate its `Error` property). In fact, we already have the correct **Exception** in the `ApiResponse<T>.Error` property, so why don't just return it?

In other words, we can think about changing the `EnsureSuccessStatusCodeAsync` method in this way:

```
public async Task<ApiResponse<T>> EnsureSuccessStatusCodeAsync()
{
    if (!IsSuccessStatusCode)
    {
        var exception = Error ?? await ApiException.Create(response.RequestMessage!, response.RequestMessage!.Method, response, Settings).ConfigureAwait(false);

        Dispose();

        throw exception;
    }

    return this;
}
```

So, if the `Error `property isn't null, we just use it, instead of recreating the exception at all.

**Comment by sciocoder** [comment]
I agree with @marcominerva 

Also check when a custom ExceptionFactory is provided, the default creates the exception if the response does not have a success status code (https://github.com/reactiveui/refit/blob/main/Refit/RefitSettings.cs#L223) and this is good for the EnsureSuccessStatusCodeAsync method of ApiResponse but if this does not happen the same disposal problem returns

https://github.com/reactiveui/refit/blob/main/Refit/RequestBuilderImplementation.cs#L261
```
     if (typeof(T) != typeof(HttpResponseMessage))
     {
         e = await settings.ExceptionFactory(resp).ConfigureAwait(false);
[+]      disposeResponse = false; // caller has to dispose
     }
```
this update would solve the disposal problem but move to the stream which is already consumed (e == null) https://github.com/reactiveui/refit/blob/main/Refit/RequestBuilderImplementation.cs#L273
